apps/api/v2/src/modules/oauth/oauth-client.controller.ts

```diff
@@ -13,65 +16,77 @@ import {
   HttpCode,
   HttpStatus,
   Logger,
-  Res,
+  UseGuards,
 } from "@nestjs/common";
+import { MembershipRole, PlatformOAuthClient } from "@prisma/client";
+
+import { SUCCESS_STATUS } from "@calcom/platform-constants";
+import { ApiResponse } from "@calcom/platform-types";
 
 @Controller({
   path: "oauth-clients",
   version: "2",
 })
+@UseGuards(NextAuthGuard, OrganizationRolesGuard)
```

Comment:
doesn't this mean that the organizationRolesGuard is applied to all the endpoints ?

Reply:
It is used together with "@Roles" decorator applied to each endpoint e.g. create has `@Roles([MembershipRole.ADMIN, MembershipRole.OWNER])` but read does not, so then the roles are passed to the `OrganizationsRolesGuard`, which checks if any roles have been passed to Roles guard or if it has been applied, and if none of them is true it just returns `true` meaning the request passes through.